src/org/exist/indexing/IndexController.java

```diff
@@ -331,6 +331,18 @@ public void indexNode(final Txn transaction, final IStoredNode node, final NodeP
     }
 
     /**
+     * Helper method: indexing is starting for a document
+     *
+     * @param transaction the current transaction
+     * @param listener the StreamListener which receives the index events
+     */
+    public void startIndexDocument(final Txn transaction, final StreamListener listener) {
```

Comment:
@shabanovd I am following the existing convention which is to pass the StreamListener as the last argument. That is fine.